src/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -905,18 +929,44 @@ public static BigInteger DivRem(BigInteger dividend, BigInteger divisor, out Big
             dividend.AssertValid();
             divisor.AssertValid();
 
-            int signNum = +1;
-            int signDen = +1;
-            BigIntegerBuilder regNum = new BigIntegerBuilder(dividend, ref signNum);
-            BigIntegerBuilder regDen = new BigIntegerBuilder(divisor, ref signDen);
-            BigIntegerBuilder regQuo = new BigIntegerBuilder();
+            bool trivialDividend = dividend._bits == null;
+            bool trivialDivisor = divisor._bits == null;
 
-            // regNum and regQuo are overwritten with the remainder and quotient, respectively
-            regNum.ModDiv(ref regDen, ref regQuo);
-            remainder = regNum.GetInteger(signNum);
-            return regQuo.GetInteger(signNum * signDen);
-        }
+            if (trivialDividend && trivialDivisor)
+            {
+                remainder = dividend._sign % divisor._sign;
+                return dividend._sign / divisor._sign;
+            }
+
+            if (trivialDividend)
+            {
+                remainder = dividend;
```

Comment:
A short comment here explaining why this is valid would be helpful.  I assume it's because we know that if it dividend is trivial and the divisor is non-trivial, then (sign excluded) the divisor must be larger than the dividend, such that the result will always be zero and the remainder will always be the dividend?